scripts/src/models/orders.ts

```diff
@@ -240,24 +240,22 @@ export const Order = {
 	 * @param      {number}  limit
 	 * @return     {Promise<T[]>}  filtered orders including p2p
 	 */
-	async getAll<T extends Order>(filters: GetOrderFilters & { userId?: string }, limit?: number): Promise<T[]> {
-		const allOrders = await this.genericGet(filters).getMany(); // can be replaced by cache
-
-		const ids: string[] = allOrders.map(order => order.id);
+	async getAll<T extends Order>(filters: GetOrderFilters & ({ userId: string } | { walletAddress: string }), limit?: number): Promise<T[]> {
+		const contexts = await OrderContext.find(filters.userId ? { userId: filters.userId } : { wallet: filters.walletAddress }); // can be replaced by cache
```

Comment:
This query is easier to cache and "should" be easier on the DB .
The filters will be applied in the following query